### ISSUE TITLE
Offline mode ignored because `ansible-galaxy role install` command does not support it

### ISSUE THREAD

**Issue by jaap3** [ansible/ansible-lint#2646]
I looked at the issue tracker, and did find some issues mentioning that offline mode was broken. But according to the most recent commit is was fixed. However I am running into the following.

##### Summary

I have a `.ansible-lint` file with these settings:

```
offline: true

mock_roles:
  - role-a
  - role-b
``` 

and a `requirements.yml` with these lines:

```
roles:
  - src: git@github.com:example/role-a.git
    scm: git
    version: v1.0.0
    name: role-a

  - src: git@github.com:example/role-b.git
    scm: git
    version: v1.2.0
    name: role-b
```

Running `ansible-lint` now suddenly prints this warning:

`WARNING  Offline mode ignored because `ansible-galaxy role install` command does not support it.`

and then attempts to install roles, this fails in CI because access to the git repositories of these roles is denied (besides, I do not want these roles to be linted in the first place).

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.13.5]
  config file = None
  configured module search path = ['~/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = ~/.pyenv/versions/3.8.12/envs/ansible/lib/python3.8/site-packages/ansible
  ansible collection location = ~/.ansible/collections:/usr/share/ansible/collections
  executable location = ~/.pyenv/versions/ansible/bin/ansible
  python version = 3.8.12 (default, Jan 27 2022, 10:53:54) [Clang 13.0.0 (clang-1300.0.29.30)]
  jinja version = 3.1.2
  libyaml = False

ansible-lint --version
ansible-lint 6.8.5 using ansible 2.13.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

macOS / Linux

##### STEPS TO REPRODUCE

Create a setup a mentioned in the summary and run `ansible-lint`

##### Desired Behavior

I would like ansible-lint to work as before, just lint my playbook and ignore the mocked roles. I am able to restore this behaviour by installing

```
ansible-compat==2.2.1
ansible-lint==6.8.4
```

##### Actual Behavior

ansible-lint attempts to instal the mocked roles, which fails

**Comment by ssbarnea** [comment]
* linter needs roles not for linting them, but for linting your content.
* mocking does limit considerably linter ability to spot errors, so it is discouraged
* as ansible has no support to install offline roles, you should take care of that step before running the linter
* the bug is valid as linter should not attempt to install the roles while in offline mode. For collections it is a bit different as galaxy command supports an offline mode for these starting with 2.14 (not for roles).

**Comment by jaap3** [comment]
wow that was quick, thanks!

**Comment by ssbarnea** [comment]
You need to wait a little bit, i need to release the entire chain to make use of it. Still, have a look at the fix, maybe you can spot an error.
